observability/traceutils.go

```diff
@@ -0,0 +1,109 @@
+package observability
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	traceapi "go.opentelemetry.io/otel/trace"
+)
+
+// CreateSpanAndLoggerFromContext creates span and logger from context with provided name.
+// Logger explicitly extended with dd.trace_id attribute for DataDog logs and traces connections
+func CreateSpanAndLoggerFromContext(ctx context.Context, tracerName, spanName string, kv ...attribute.KeyValue) (context.Context, traceapi.Span, zerolog.Logger) {
```

Comment:
If you really wanna go this route then something like this would make sense though personally I might be overkill: 
```
type Observability struct {
 logger
 tracer
}

NewObservability(ctx, tracer, span, ...kv)

func (o Observability) RecordEvent(event...)
func (o Observability) RecordError(error...)
func (o Observability) SetAttributes(kv...)
```

Reply:
1. This function actually creates a logger from the existing context, so this API is available to use inside the code if necessary.
2. Not sure what you mean here. But if I understand you right logger and span are available when using this utils functions, so when necessary you still can use standard API since this is just no more then an instrument 
3. I do not see the necessity to do this, we would need to pass this structure everywhere, while we just do not need to store this state. We have context everywhere, which is a pretty default pattern + traces is available globally.

Reply:
I don't like how it looks. 
And again when something says it does 2 things it probably shouldn't.

Reply:
Right, but two different, unrelated ,things. But in our case this instrument do only actions related to observability, logs, and traces.